### PR TITLE
Switch default NL writer to nlv2

### PR DESCRIPTION
**Pull request by jsiirola**
## Summary/Motivation:
This switches the default NL writer to the new "NLv2" writer.

## Changes proposed in this PR:
- Change default NL writer
- Update `pyomo.mpec.complementarity` to suppress deprecation warnings
- Testing
  - Add explicit NL v1 and v2 tests for complementarity and suffixes
  - Skip a test on old versions of CPLEX where changing variable ordering could cause CPLEX to calculate incorrect duals
- Fix handling of the default `file_determinism` flag int eh NLv1 writer
- NLv2 writer updates
  - NL writer will by default write files using Unix-style (not universal) newlines.  This is due to a bug in how the ASL handles string arguments (it cannot correctly parse string arguments terminated with windows-style newlines)
  - map `inf`/`-inf` bounds on Constraints/Vars to `None`
  - Add warnings for writing NL files containing string arguments using `\r\n` line terminators
  - Add processing of complementarity constraints (not the preferred solution, but this replicated the functionality in the NLv1 writer)

### Legal Acknowledgement

By contributing to this software project, I have read the [contribution guide](https://pyomo.readthedocs.io/en/stable/contribution_guide.html) and agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the BSD license.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Comment by michaelbynum** [comment]
We should add an issue to eventually remove the old nl writer.

File: pyomo/scripting/solve_config.py [hunk 1]
```diff
@@ -107,7 +107,7 @@ def minlp_config_block(init=False):
         None,
     )).declare_as_argument(dest='symbolic_solver_labels')
     model.declare('file determinism', ConfigValue(
-        1,
+        None,
         int,
         'When interfacing with a solver using file based I/O, set '
         'the effort level for ensuring the file creation process is '
```
Comment (michaelbynum) [inline pyomo/scripting/solve_config.py:113]: Can you update this doc string?

**Review by michaelbynum — APPROVED** [review]
One minor comment below.

**Review by mrmundt — APPROVED** [review]
Looks really good, @jsiirola . I didn't even see any typos.

**Pull request merged**